libvast_test/src/main.cpp

```diff
@@ -82,32 +78,41 @@ extern std::set<std::string> config;
 
 } // namespace vast::test
 
-int main(int argc, char** argv) {
+namespace {
+std::vector<std::string> get_test_args(int argc, const char* const* argv) {
   // Parse everything after after '--'.
-  std::string delimiter = "--";
-  auto start = argc;
-  for (auto i = 1; i < argc - 1; ++i) {
-    if (argv[i] == delimiter) {
-      start = i + 1;
-      break;
-    }
-  }
+  constexpr std::string_view delimiter = "--";
+  auto start = argv + 1;
+  auto end = argv + argc;
+  auto args_start = std::find(start, end, delimiter);
+  if (args_start == end)
+    return {};
+  return {args_start + 1, end};
+}
+} // namespace
+
+int main(int argc, char** argv) {
   std::string vast_loglevel = "quiet";
-  if (start != argc) {
-    auto res
-      = caf::message_builder(argv + start, argv + argc)
-          .extract_opts({
-            {"vast-verbosity", "console verbosity for libvast", vast_loglevel},
-          });
-    if (!res.error.empty()) {
-      std::cout << res.error << std::endl;
+  auto test_args = get_test_args(argc, argv);
+  if (!test_args.empty()) {
+    auto options
+      = caf::config_option_set{}
+          .add(vast_loglevel, "vast-verbosity", "console verbosity for libvast")
+          .add<bool>("help", "print this help text");
+    caf::settings cfg;
+    auto res = options.parse(cfg, test_args);
+    if (res.first != caf::pec::success) {
+      std::cout << "error while parsing argument \"" << *res.second
+                << "\": " << to_string(res.first) << "\n\n";
+      std::cout << options.help_text() << std::endl;
       return 1;
     }
-    if (res.opts.count("help") > 0) {
-      std::cout << res.helptext << std::endl;
+    if (caf::get_or(cfg, "help", false)) {
+      std::cout << options.help_text() << std::endl;
       return 0;
     }
-    vast::test::config = std::move(res.opts);
+    vast::test::config = {make_move_iterator(begin(test_args)),
+                          make_move_iterator(end(test_args))};
```

Comment:
Please qualify calls into namespace std instead of ADL-using `make_move_iterator`, `begin`, and `end` here.

Reply:
My motivation here is to have a code that doesn't need any change if we introduce a 3rd party container instead of STL one.
It could provide it's own begin/end free function.

Although my approach has one pitfall that i have just recently read.
Let's say the 3rd party container has .begin() and .end() method.

Then std::begin(x) would return x-.begin(), whereas begin(x) will lead to compilation error.

The solution would be to use using std::begin, but this seems like too much for a code that won't probably use non STL container.

I will qualify into std :D